domains/auth/Auth.API/API/Controllers/LoginController.cs

```diff
@@ -28,4 +51,92 @@ public LoginController(IOidcService oidcService)
 
         return oidcService.CreateAuthorizationUri(state);
     }
+
+    [HttpGet]
+    [Route("/oidc/login/callback")]
+    public async Task<ActionResult<NextStep>> CallbackAsync(
+        OidcCallbackParams oidcCallbackParams,
+        [FromServices] ICryptographyFactory stateCryptography,
+        [FromServices] ICryptographyFactory IdtokenCryptography
+    )
+    {
```

Comment:
This is a very long method indicating that it does several things. Would it make sense to split it into several methods (https://refactoring.com/catalog/extractFunction.html))